### DISCUSSION TITLE
Are resource accounts no longer considered helpful since the launch of Objects?

### DISCUSSION THREAD

**Discussion by magnum6actual** [aptos-labs/aptos-developer-discussions#454] [General]
Prior to Objects.move, we used resource accounts for a lot of things. After Objects launched, I would always see references (mostly on Discord) recommending to use objects instead of resource accounts.  Are there no longer "best practices" use cases for resource accounts?

**Comment by MoonShiesty** [comment]
One use case for deploying code to resource accounts is that resource account can store a `SignerCap` and generate a signer for the resource account

Code objects restrict signing to upgrading the code

**Reply by gregnazario** [reply]
Most cases can be replaced with Objects and with some benefits and tradeoffs:

1. Objects have automatic handling of upgrades, where Resource Accounts, you need to handle upgrades directly in your contract, often having to copy this logic between different deployed contracts.
2. Resource accounts allow you to use the SignerCap to get the direct signer that handles the code, where objects requires you to control who owns the object to allow for upgrading.  This is a little more flexible on objects, because you can change the logic by determining who controls the object.
3. Resource accounts let you colocate data with the code, by using the `SignerCap` directly with the object.

Generally, I would say though it's easier to use an object and not have to handle putting the `SignerCap` and upgrade logic directly in your contract. It could cause some confusion later when you can't upgrade your contract because you missed this.
